### PR TITLE
Initial version of the SNMP integration plugin, incorporating snmp and snmptrap inputs

### PR DESCRIPTION
**Pull request by edmocosta**
## What does this PR do?

Created the SNMP Integration Plugin, incorporating the [logstash-input-snmp](https://github.com/logstash-plugins/logstash-input-snmp) and [logstash-input-snmptrap](https://github.com/logstash-plugins/logstash-input-snmptrap) plugins.

- No code changes were done, only small fixes, adjustments and project structure moves.
- A flaky test was [fixed](https://github.com/logstash-plugins/logstash-integration-snmp/pull/1/commits/04bef44be3621a57a2273d2f2eec46567e4a8af7#diff-187a09cb98206d777155af728904aecb3967b1c77bedeb36b6b3c4d0d5f8ecb2R408) and the `.ci` adjusted to work with both plugins. It was mainly adapted from the original CI configurations.
- Unit tests and integration tests were placed into two separate folder `spec/integration` and `spec/unit`. The specs were not modified.
- Docs were adjusted to follow the integration plugins pattern: 
  - index.asciidoc: summarizes the provided plugins
  - input-snmp.asciidoc: no changes from the original input plugin but the header `:type: integration` markups
  - input-snmptrap.asciidoc: no changes from the original input plugin but the header `:type: integration` markups

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by roaksoax — APPROVED** [review]
I've gone through all of the plugin, given that the code is the same, and it's only organizational, I haven't gone through a thorough review. But LGTM!

File: docs/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+:include_path: ../../../../logstash/docs/include
+///////////////////////////////////////////
+END - GENERATED VARIABLES, DO NOT EDIT!
+///////////////////////////////////////////
+
+[id="plugins-{type}s-{plugin}"]
+
+=== SNMP Integration Plugin
+
+include::{include_path}/plugin_header.asciidoc[]
```
Comment (karenzone) [inline docs/index.asciidoc:20]: The `plugin_header.asciidoc` file include can stay.  Technically, the integration plugin itself (but not its component plugins) behaves like a traditional (input, output, codec, filter) plugin.  Only the `plugin_header-integration.asciidoc` header file causes problems.

**Review by karenzone — COMMENTED** [review]
@edmocosta, thank you for the opportunity to try to get these new integration docs working from the beginning. I've commented on some lines that I know will cause problems if unresolved, and added explanations.  Please let me know if you'd like to discuss. 

Let's check in again before you merge and publish.

**Comment by karenzone** [comment]
@edmocosta Thank you for the opportunity to work through adding an integration iteratively.  We might still hit some bumps, but we'll learn the best practices and then I'll document them for next time.

**Pull request merged**